File: public/public-persistence-jooq/src/main/java/ch/difty/scipamato/publ/persistence/keyword/JooqKeywordRepo.java
package ch.difty.scipamato.publ.persistence.keyword;

import static ch.difty.scipamato.common.persistence.TranslationUtilsKt.trimLanguageCode;
import static ch.difty.scipamato.publ.db.tables.Keyword.KEYWORD;

import java.util.List;

import org.jetbrains.annotations.NotNull;
import org.jooq.DSLContext;
import org.springframework.stereotype.Repository;

import ch.difty.scipamato.publ.entity.Keyword;

@Repository
public class JooqKeywordRepo implements KeywordRepository {

    private final DSLContext dslContext;

    public JooqKeywordRepo(@NotNull final DSLContext dslContext) {
        this.dslContext = dslContext;
    }

    @NotNull
    @Override
    public List<Keyword> findKeywords(@NotNull final String languageCode) {
        final String lang = trimLanguageCode(languageCode);
        // skipping the audit fields
        return dslContext
            .select(KEYWORD.ID, KEYWORD.KEYWORD_ID, KEYWORD.LANG_CODE, KEYWORD.NAME, KEYWORD.SEARCH_OVERRIDE)
            .from(KEYWORD)
            .where(KEYWORD.LANG_CODE.eq(lang))
            .orderBy(KEYWORD.NAME)
            .fetchInto(Keyword.class);
    }
}

File: public/public-persistence-jooq/src/main/kotlin/ch/difty/scipamato/publ/persistence/keyword/JooqKeywordRepo.kt
package ch.difty.scipamato.publ.persistence.keyword

import ch.difty.scipamato.common.persistence.trimLanguageCode
import ch.difty.scipamato.publ.entity.Keyword
import org.jooq.DSLContext
import org.springframework.stereotype.Repository

@Repository
open class JooqKeywordRepo(private val dslContext: DSLContext) : KeywordRepository {

    @Suppress("MagicNumber")
    override fun findKeywords(languageCode: String): List<Keyword> = dslContext
        .select(
            ch.difty.scipamato.publ.db.tables.Keyword.KEYWORD.ID,
            ch.difty.scipamato.publ.db.tables.Keyword.KEYWORD.KEYWORD_ID,
            ch.difty.scipamato.publ.db.tables.Keyword.KEYWORD.LANG_CODE,
            ch.difty.scipamato.publ.db.tables.Keyword.KEYWORD.NAME,
            ch.difty.scipamato.publ.db.tables.Keyword.KEYWORD.SEARCH_OVERRIDE
        )
        .from(ch.difty.scipamato.publ.db.tables.Keyword.KEYWORD)
        .where(ch.difty.scipamato.publ.db.tables.Keyword.KEYWORD.LANG_CODE.eq(trimLanguageCode(languageCode)))
        .orderBy(ch.difty.scipamato.publ.db.tables.Keyword.KEYWORD.NAME)
        .map { r -> Keyword(r[0] as Int, r[1] as Int, r[2] as String?, r[3] as String?, r[4] as String?) }
}
